tools/src/main/scala/scala/scalanative/optimizer/pass/AllocLowering.scala

```diff
@@ -21,8 +23,15 @@ class AllocLowering(implicit fresh: Fresh, top: Top) extends Pass {
 
     insts.foreach {
       case Let(n, Op.Classalloc(ClassRef(cls))) =>
-        val size = let(Op.Sizeof(cls.layout.struct))
-        let(n, Op.Call(allocSig, alloc, Seq(cls.rtti.const, size), Next.None))
+        val size = MemoryLayout.sizeOf(cls.layout.struct)
+        val allocMethod =
+          if (size < LARGE_OBJECT_MIN_SIZE) alloc else largeAlloc
+
+        let(n,
+            Op.Call(allocSig,
+                    allocMethod,
+                    Seq(cls.rtti.const, Val.Long(size)),
+                    Next.None))
```

Comment:
Now that I think about it, allocation helpers are never used anywhere apart from the array implementation. We should get rid off them.